demos/random-values-stream/index.js

```diff
@@ -0,0 +1,217 @@
+            console.error(error);
+
+            this.clearValueInterval();
+        },
+
+        startValueInterval(controller) {
+            if (this.interval) {
+                return;
+            }
+
+            this.interval = setInterval(() => {
+                try {
+                    controller.enqueue(randomValuesUint8Array(20));
+                    this.totalEnqueuedItemsCount++;
+                    this.checkBackpressureSignal(controller);
+                    
+                    // Close the stream and clear the interval
+                    if (maxValues && this.totalEnqueuedItemsCount >= maxValues) {
+                        return this.close(controller);  
+                    }
+                } catch (error) {
+                    this.throwFinalError(error);
+                }
+            }, valueInterval);
+        },
+        
+        /**
+         * Clears the value interval stored in this.interval reference
+         */
+        clearValueInterval() {
+            if (this.interval) {
+                clearInterval(this.interval);
+                this.interval = null;
+            }
+        },
+
+        /**
+         * Checks a backpressure signal  and clears the interval
+         * not enqueuing any more values
+         * 
+         * @param {*} controller 
+         * @param {*} interval
+         */
+        checkBackpressureSignal(controller, interval) {
+            if (controller.desiredSize <= 0) {
+                this.clearValueInterval();
+            }
+        }
+    }, cqs);
+
+    return readableStream;
+}
+
+/**
+ * Creates a random values Uint8Array
+ * @param {*} numberOfBytes 
+ */
+function randomValuesUint8Array(numberOfBytes) {
+    const uint8Array = new Uint8Array(numberOfBytes);
+    return window.crypto.getRandomValues(uint8Array);
+}
+
+/**
+ * Create a writable stream to display the output with a builtin backpressure
+ */
+function createOutputWritableStream(parentElement) {
+    /**
+     * Equivalent to
+     * 
+     * const cqs = new CountQueuingStrategy({
+     *  highWaterMark: 3,
+     * });
+     */
+    const queuingStrategy = {
+        highWaterMark: 3,
+        size() { return 1; }
+    }
+
+    const writable = new WritableStream({
+        async write(chunk, controller) {
+            try {
+                await writeChunk(chunk);
+                return;
+            } catch (error) {
+                return this.finalErrorHandler(error, controller);
+            }
+        },
+
+        finalErrorHandler(error, controller) {
+            controller.error(error);
+            logStatusText('Error occured in the writable stream');
+            return error;
+        },
+
+        close() {
+            logStatusText('Closing the stream');
+            console.log('Stream closed');
+        }
+    });
+
+    /**
+     * Writes a chunk to the span and appends it to the parent element
+     * @param {*} chunk 
+     */
+    async function writeChunk(chunk) {
```

Comment:
This function doesn't need to be async as it performs no asynchronous operations. Also, as it is only called within write(), it might as well just be included in that function.